### PR TITLE
Remove old Link branding

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Removes old Link branding.

## Motivation
https://jira.corp.stripe.com/browse/RUN_MOBILESDK-3108

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift [hunk 1]
```diff
@@ -91,7 +91,7 @@ extension STPPaymentMethod {
         case .SEPADebit:
             return Image.carousel_sepa.makeImage().withRenderingMode(.alwaysOriginal)
         case .link:
-            return (LinkUI.useNewBrand ? Image.link_logo : Image.pm_type_link_deprecated).makeImage().withRenderingMode(.alwaysOriginal)
+            return Image.link_logo.makeImage()
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:94]: There is a few in this PR where we did not need `makeImage()` previously but now we do. I'm not seeing how this complied before..

Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:94]: This was one expression:
`(LinkUI.useNewBrand ? Image.link_logo : Image.pm_type_link_deprecated)`
Which then had `.makeImage()` called on it. Sorry for the ambiguity!

Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:94]: You should keep `.withRenderingMode(.alwaysOriginal)`, we don't want it to render as a template.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PayWithLinkButton.swift [hunk 2]
```diff
@@ -144,7 +138,7 @@ final class PayWithLinkButton: UIControl {
     private lazy var payWithStackView: UIStackView = {
         let stackView = UIStackView(arrangedSubviews: [
             payWithLinkView,
-            LinkUI.useNewBrand ? nil : Self.makeDeprecatedArrowView(),
+            nil,
```
Comment (davidme-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PayWithLinkButton.swift:141]: The `nil`s can be removed from the stackview.

**Review by davidme-stripe — CHANGES_REQUESTED** [review]
Thanks! Added a few notes

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift (hunk 1 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentOption+Images.swift:94]: Oh I missed the parenthesis! Thanks!

**Review by davidme-stripe — APPROVED** [review]

**Pull request merged**